In [6]:
# Question 2.15

A <- matrix(c(5,2,3,4,-3,7,4,1,2), 3, 3)
B <- matrix(c(1,0,1,0,1,2,1,0,3), 3, 3)
det(B)

[1] 2

In [14]:
# Question 2.16

3 * (72 - 36) - 4 * (36 - 18)

T <- matrix(c(sqrt(3), 0, 0, 4 / sqrt(3), sqrt(8 - 16/3), 0, 3 / sqrt(3), 2 / (8 - 16 /3), sqrt(9 - 3 - 4 /(8 - 16/3))), 3, 3)
t(T) %*% T


[1] 36

[,1] [,2]     [,3]    
[1,] 3    4.000000 3.000000
[2,] 4    8.000000 5.224745
[3,] 3    5.224745 8.062500

In [17]:
# Question 2.17

T <- matrix( c( sqrt(3), 0, 0, - 5/sqrt(3), sqrt(14/3), 0, -1/sqrt(3), -5/(3*sqrt(14/3)), sqrt(1- 75/126 - 1/3)), 3, 3)

t(T) %*% T

[,1] [,2]          [,3]         
[1,]  3   -5.000000e+00 -1.000000e+00
[2,] -5    1.300000e+01  3.001531e-16
[3,] -1    3.001531e-16  1.000000e+00

In [20]:
# 2.18
C <- matrix(c( 1 / sqrt(6), 2 / sqrt(6), 1 / sqrt(6), -1/sqrt(3), 1/sqrt(3), -1/sqrt(3), 1/sqrt(2), 0, 1/sqrt(2)),3,3)
C

[,1]      [,2]       [,3]     
[1,] 0.4082483 -0.5773503 0.7071068
[2,] 0.8164966  0.5773503 0.0000000
[3,] 0.4082483 -0.5773503 0.7071068

In [21]:
# 2.19

x_1 <- matrix(c(1/sqrt(18), 4/sqrt(18), -1/sqrt(18)), 3, 1)
x_1

[,1]      
[1,]  0.2357023
[2,]  0.9428090
[3,] -0.2357023

In [5]:
# Question 2.21 part a
A = matrix(c(2, -1, -1, 2), 2, 2)
print(A)
print(eigen(A))
1 / sqrt(2)

     [,1] [,2]
[1,]    2   -1
[2,]   -1    2
eigen() decomposition
$values
[1] 3 1

$vectors
           [,1]       [,2]
[1,] -0.7071068 -0.7071068
[2,]  0.7071068 -0.7071068



[1] 0.7071068

In [20]:
# Question 2.21 part b
# FIXME - Not correct
C = matrix(c( -1 / sqrt(2), 1 / sqrt(2), -1 / sqrt(2), -1 / sqrt(2)), 2, 2)
D12 = matrix(c(sqrt(3),0,0,sqrt(1)), 2, 2)
A12 = C * D12 * t(C)
print(A12 * A12)

     [,1] [,2]
[1,] 0.75 0.00
[2,] 0.00 0.25


In [2]:
A = matrix(c(5,5,5,5,5,
             5,5,5,5,5,
             5,5,5,5,5,
             5,5,5,5,5,
             5,5,5,5,5), 5,5)
print(A)
print(eigen(A))

     [,1] [,2] [,3] [,4] [,5]
[1,]    5    5    5    5    5
[2,]    5    5    5    5    5
[3,]    5    5    5    5    5
[4,]    5    5    5    5    5
[5,]    5    5    5    5    5
eigen() decomposition
$values
[1] 2.500000e+01 7.105427e-15 2.775558e-16 1.387779e-16 9.016384e-34

$vectors
           [,1]       [,2]       [,3]       [,4]          [,5]
[1,] -0.4472136  0.8944272  0.0000000  0.0000000  0.000000e+00
[2,] -0.4472136 -0.2236068  0.7071068 -0.5000000 -4.976278e-17
[3,] -0.4472136 -0.2236068  0.2357023  0.8333333 -1.658759e-17
[4,] -0.4472136 -0.2236068 -0.4714045 -0.1666667 -7.071068e-01
[5,] -0.4472136 -0.2236068 -0.4714045 -0.1666667  7.071068e-01



# Chapter 3

In [13]:
# Question 3.10
n = 10
Y <- matrix(c(35,35,40,10,6,20,35,35,35,30,
                 3.5,4.9,30,2.8,2.7,2.8,4.6,10.9,8,1.6,
                 2.8,2.7,4.38,3.21,2.73,2.81,2.88,2.9,3.28,3.2), 10,3)
I <- diag(10)
C <- matrix(ncol = 10, nrow = 10)
C[ , ] <- 1/n
Y
I
C


[,1] [,2] [,3]
 [1,] 35    3.5 2.80
 [2,] 35    4.9 2.70
 [3,] 40   30.0 4.38
 [4,] 10    2.8 3.21
 [5,]  6    2.7 2.73
 [6,] 20    2.8 2.81
 [7,] 35    4.6 2.88
 [8,] 35   10.9 2.90
 [9,] 35    8.0 3.28
[10,] 30    1.6 3.20

[,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
 [1,] 1    0    0    0    0    0    0    0    0    0    
 [2,] 0    1    0    0    0    0    0    0    0    0    
 [3,] 0    0    1    0    0    0    0    0    0    0    
 [4,] 0    0    0    1    0    0    0    0    0    0    
 [5,] 0    0    0    0    1    0    0    0    0    0    
 [6,] 0    0    0    0    0    1    0    0    0    0    
 [7,] 0    0    0    0    0    0    1    0    0    0    
 [8,] 0    0    0    0    0    0    0    1    0    0    
 [9,] 0    0    0    0    0    0    0    0    1    0    
[10,] 0    0    0    0    0    0    0    0    0    1

[,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
 [1,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [2,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [3,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [4,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [5,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [6,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [7,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [8,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
 [9,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  
[10,] 0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1  0.1

In [15]:
Sigma <- (n-1)^(-1) * (t(Y) %*% (I - C) %*% Y)
Sigma

[,1]       [,2]      [,3]     
[1,] 140.544444 49.680000 1.9412222
[2,]  49.680000 72.248444 3.6760889
[3,]   1.941222  3.676089 0.2501211

# Chapter 5

In [5]:
qchisq(0.05, 2, lower.tail = FALSE)

[1] 5.991465

In [9]:
(164.7-170)/ (sqrt(1000/20))

[1] -0.7495332

In [11]:
qnorm(0.05/2)

[1] -1.959964

**Question 5.12**  
Use the probe data in Table 3.6.

In [3]:
Y <- matrix(c(51,27,37,42,27,43,41,38,36,26,29,
                 36,20,22,36,18,32,22,21,23,31,20,
                 50,26,41,32,33,43,36,31,27,31,25,
                 35,17,37,34,14,35,25,20,25,32,26,
                 42,27,30,27,29,40,38,16,28,36,25), 11,5)
Y


[,1] [,2] [,3] [,4] [,5]
 [1,] 51   36   50   35   42  
 [2,] 27   20   26   17   27  
 [3,] 37   22   41   37   30  
 [4,] 42   36   32   34   27  
 [5,] 27   18   33   14   29  
 [6,] 43   32   43   35   40  
 [7,] 41   22   36   25   38  
 [8,] 38   21   31   20   16  
 [9,] 36   23   27   25   28  
[10,] 26   31   31   32   36  
[11,] 29   20   25   26   25

a) Test $H_0: \mu = (30,25,40,25,30)'$

In [41]:
# Setup
n = 11
mu_0 <- matrix(c(30,25,40,25,30), 5,1)
y_bar <- matrix(nrow = 5, ncol =1)
for ( i in 1:5) {
    y_bar[i,] <- sum(Y[,i] / n)
}
J <- matrix(nrow = 11, ncol = 11)
J[,] = 1
I <- diag(11)
S <- (t(Y) %*% (I - (1/n) * J) %*% Y) / (n-1)

# Test Statistic
T2 <- n * t((y_bar - mu_0)) %*% solve(S) %*% (y_bar - mu_0)
T2

[,1]   
[1,] 85.3327

$85.33 > 36.56 \text{ from Appendix A.7 } \therefore$ reject $H_0$.

b) If null is rejected, test each variable seperately, using (5.3)

In [42]:
# Make a vector containing the 5 individual t-values.
t <- matrix(nrow = 5, ncol =1)
for (i in 1:5) {
    t[i,] = (y_bar[i,] - mu_0[i,])  / (S[i,i] / sqrt(n))
}
t

[,1]       
[1,]  0.31035455
[2,]  0.03926549
[3,] -0.32291883
[4,]  0.11999366
[5,]  0.04143191

We fail to reject the null in every one of these cases.

**Question 5.13**  
For the prove word data in Table 3.6, test $H_0: \mu_1 = \mu_2 = ... = \mu_5$, using $T^2$ in (5.33).  
Skipped for now.

**Question 5.14**  
Use the ramus bone data in Table 3.7:

In [46]:
Y <- matrix(c(47.8,46.4,46.3,45.1,47.6,52.5,51.2,49.8,48.1,45.0,51.2,48.5,52.1,48.2,49.6,50.7,47.2,53.3,46.2,46.3,
              48.8,47.3,46.8,45.3,48.5,53.2,53.0,50.0,50.8,47.0,51.4,49.2,52.8,48.9,50.4,51.7,47.7,54.6,47.5,47.6,
              49.0,47.7,47.8,46.1,48.9,53.3,54.3,50.3,52.3,47.3,51.6,53.0,53.7,49.3,51.2,52.7,48.4,55.1,48.1,51.3,
              49.7,48.4,48.5,47.2,49.3,53.7,54.5,52.7,54.4,48.3,51.9,55.5,55.0,49.8,51.8,53.3,49.5,55.3,48.4,51.8), nrow = 20, ncol =4)

a) Test $H_0: \mu = (48,49,50,51)'$.

In [50]:
# Calculations
n <- 20
mu_0 <- matrix(c(48,49,50,51), 4,1)
y_hat <- matrix(nrow = 4, ncol =1)
for (i in 1:4) {
    y_hat[i,] = sum(Y[,i]) / n
}
J <- matrix(nrow = 20, ncol = 20)
J[,] = 1
I <- diag(20)
S <- (t(Y) %*% (I - (1/n) * J) %*% Y) / (n-1)

# Test Statistic
T2 <- n * t((y_hat - mu_0)) %*% solve(S) %*% (y_hat - mu_0)
T2

[,1]    
[1,] 1.792636

b) If $H_0$ rejected, test each variable separately, using (5.3)

Not significant.

**Question 5.15**  
Skip for now, Profile Analysis.

**Question 5.16**  
Four measurements were made on two species of flea beetles (Lubischew 1962). The tables were...

In [5]:
Haltica_Oleracea <- matrix(c(189,192,217,221,171,192,213,192,170,201,195,205,180,192,200,192,200,181,192,
                             245,260,276,299,239,262,278,255,244,276,242,263,252,283,294,277,287,255,287,
                             137,132,141,142,128,147,136,128,128,146,128,147,121,138,138,150,136,146,141,
                             163,217,192,213,158,173,201,185,192,186,192,192,167,183,188,177,173,183,198), nrow = 19, ncol = 4)

Haltica_Carduorum <- matrix(c(181,158,184,171,181,181,177,198,180,177,176,192,176,169,164,181,192,181,175,197,
                              305,237,300,273,297,308,301,308,286,299,317,312,285,287,265,308,276,278,271,303,
                              184,133,166,162,163,160,166,141,146,171,166,166,141,162,147,157,154,149,140,170,
                              209,188,231,213,224,223,221,197,214,192,213,209,200,214,192,204,209,235,192,205), nrow = 20, ncol = 4)

a) Test $H_0: \mu_1 = \mu_2$ using $T^2$.

In [1]:
# Preliminary Calculations

n1 = 19
n2 = 20
J1 <- matrix(nrow = 19, ncol = 19)
J2 <- matrix(nrow = 20, ncol = 20)
J1[,] = 1
J2[,] = 1
I_19 <- diag(19)
I_20 <- diag(20)
mu1 <- matrix(nrow = 4, ncol = 1)
for (i in 1:4) {
    mu1[i,] = sum(Haltica_Oleracea[,i] / n1)
}
mu2 <- matrix(nrow = 4, ncol = 1)
for (i in 1:4) {
    mu2[i,] = sum(Haltica_Carduorum[,i] / n2)
}

S1 <- (t(Haltica_Oleracea) %*% (I_19 - (1 / n1) * J1) %*% Haltica_Oleracea) / (n1 -1)
S2 <- (t(Haltica_Carduorum) %*% (I_20 - (1 / n2) * J2) %*% Haltica_Carduorum) / (n2 -1)
Spl <- ((n1 - 1) * S1 + (n2 - 1) * S2) / (n1 + n2 - 2)

# Test Statistic
t_obs = (n1 * n2 / (n1 + n2)) * t(mu1 - mu2) %*% solve(Spl) %*% (mu1 - mu2)
t_obs


ERROR: Error in eval(expr, envir, enclos): object 'Haltica_Oleracea' not found


b) If the $T^2$-test in part a) rejects $H_0$, carry out a t-test on each variable, as in (5.15).

In [57]:
S2pl <- matrix(nrow = 4, ncol = 1)
for ( i in 1:4) {
    S2pl[i,] = ( (n1-1) * S1[i,i] + (n2-1) * S2[i,i] ) / (n1 + n2 -2)
}
 
t <- matrix(nrow = 4, ncol = 1)
for (i in 1:4) {
    t[i,] = (mu1[i,] - mu2[i,]) / sqrt(S2pl[i,] * (1/n1 + 1/n2))
}
t

[,1]     
[1,]  3.887946
[2,] -3.865239
[3,] -5.691131
[4,] -5.042625

c) Calculate the discriminant function coefficient vector $\textbf{a} = \textbf{S}_{pl}^{-1} (\bar{\textbf{y}}_1 - \bar{\textbf{y}}_2)$

In [62]:
a <- solve(Spl) %*% (mu1 - mu2)
x <- t(a) %*% t(Y)

[,1]      
[1,]  0.3452490
[2,] -0.1303878
[3,] -0.1064338
[4,] -0.1433533

Define random variable $\textbf{x} = \textbf{a}'\textbf{y}$, with $\textbf{y} = (y_1,y_2,y_3,y_4)'$. Let $\mu_x = E(x)$. Is $\mu_x$ different for the two species? Use a test to state your conclusion.

**Question 5.21**  
Each of 15 students wrote an informal and a formal essay Kramer (1972, p. 100). The variables recorded were the number of words and the number of verbs:  
$y_1 =$ Number of words in the informal essay  
$y_2 =$ Number of verbs in the informal essay  
$x_1 =$ Number of words in the formal essay  
$x_2 =$ Number of verbs in the formal essay

In [63]:
Informal <- matrix( c(148,159,144,103,121, 89,119,123, 76,217,148,151, 83,135,178,
                       20, 24, 19, 18, 17, 11, 17, 13, 16, 29, 22, 21,  7, 20, 15), nrow = 15, ncol = 2)

Formal <- matrix( c(137,164,224,208,178,128,154,158,102,214,209,151,123,161,175,
                     15, 25, 27, 33, 24, 20, 18, 16, 21, 25, 24, 16, 13, 22, 23), nrow = 15, ncol = 2)

a) Test $\mu_d =0$.

In [84]:
# Preliminary Calculations
n=15
I <- diag(15)
J <- matrix(nrow = 15, ncol = 15)
J[,] = 1

d <- matrix(nrow = 15, ncol =2)
for (i in 1:2) {
    for(j in 1:15) {
        d[j,i] = Informal[j,i] - Formal[j,i]
    }
}
d_bar <- matrix(c(sum(d[,1]/n), sum(d[,2]/n)), nrow = 2, ncol =1)
S_d <- t(d) %*% (I - (1/n) * J) %*% d / (n-1)

# Test Statistic
T2_obs <- n * t(d_bar) %*% solve(S_d) %*% d_bar
T2_obs




[,1]    
[1,] 15.19123

In [80]:
d_bar

[,1]      
[1,] -32.800000
[2,]  -3.533333

In [79]:
d

[,1] [,2]
 [1,]   11   5 
 [2,]   -5  -1 
 [3,]  -80  -8 
 [4,] -105 -15 
 [5,]  -57  -7 
 [6,]  -39  -9 
 [7,]  -35  -1 
 [8,]  -35  -3 
 [9,]  -26  -5 
[10,]    3   4 
[11,]  -61  -2 
[12,]    0   5 
[13,]  -40  -6 
[14,]  -26  -2 
[15,]    3  -8

# Chapter 6

**Question 6.27**  
Baten, Tack, and Baeder (1958) compared judges' scores on fish prepared by three methods. Twelve fish were cooked by each method, and several judges tasted fish samples and rated each on four variables:  
  
$y_1 =$ aroma,  
$y_2 =$ flavor,  
$y_3 =$ texture,  
$y_4 =$ moisture  

Each entry is an aveage score for the judges on that fish.

## Question 5 Chat GPT

In [2]:
simulate_type_I_error <- function(p, n) {
  # Simulate the data under the null hypothesis
  reject_LR <- 0
  reject_u_prime <- 0
  
  for (i in 1:1000) {
    Y <- matrix(rnorm(n*p), nrow=n)
    S <- cov(Y)
    
    # Calculate test statistics
    u <- p^2 * det(S) / (trace(S))^p
    LR <- -2 * log(u) - n * log(u)
    u_prime <- (n - 1 - (2*p^2 + p + 2) / 6*p) * log(u)

    # Critical value for chi-squared distribution with p*(p+1)/2 df at 5% significance level
    critical_val <- qchisq(0.95, p*(p+1)/2)

    

    if (LR > critical_val) {

      reject_LR <- reject_LR + 1

    }

        if (u_prime > critical_val) {

      reject_u_prime <- reject_u_prime + 1

    }

  }

  

  c(reject_LR/1000, reject_u_prime/1000)

}

In [5]:
p = 3

n_values = c(10, 20, 30, 40, 500, 1000)

results_a <- sapply(n_values, simulate_type_I_error(p, 10))

ERROR: Error in .primTrace(what): argument must be a function


# Chapter 8 

**8.7** For the phsychological data in Table 5.1, the discriminant function coefficient vector was given in Example 5.5.  
  
**a)** Find the standardized coefficients.

In [15]:
# 8,7
males <- matrix(c(15,17,15,13,20,15,15,13,14,17,17,17,15,18,18,15,18,10,18,18,13,16,11,16,16,18,16,15,18,18,17,19,
                  17,15,14,12,17,21,13,05,07,15,17,20,15,19,18,14,17,14,21,21,17,16,15,13,13,18,15,16,19,16,20,19,
                  24,32,29,10,26,26,26,22,30,30,26,28,29,32,31,26,33,19,30,34,30,16,25,26,23,34,28,29,32,33,21,30,
                  14,26,23,16,28,21,22,22,17,27,20,24,24,28,27,21,26,17,29,26,24,16,23,16,21,24,27,24,23,23,21,28),
                  nrow = 32, ncol = 4)
femal <- matrix(c(13,14,12,12,11,12,10,10,12,11,12,14,14,13,14,13,16,14,16,13,02,14,17,16,15,12,14,13,11,07,12,06,
                  14,12,19,13,20,09,13,08,20,10,18,18,10,16,08,16,21,17,16,16,06,16,17,13,14,10,17,15,16,07,15,05,
                  12,14,21,10,16,14,18,13,19,11,25,13,25,08,13,23,26,14,15,23,16,22,22,16,20,12,24,18,18,19,07,06,
                  21,26,21,16,16,18,24,23,23,27,25,26,28,14,25,28,26,14,23,24,21,26,28,14,26,09,23,20,28,18,28,13),
                  nrow = 32, ncol = 4)
colnames(males) <- c("y1", "y2", "y3", "y4")
colnames(femal) <- c("y1", "y2", "y3", "y4")
n=32
J <- function(n) {
    matrix = matrix(1, nrow = n, ncol = n)
    return(matrix)}

$ z = \textbf{a}'\textbf{y}$  
where $\textbf{a} = \textbf{S}_{pl}^{-1} (\bar{\textbf{y}_1} - \bar{\textbf{y}_2})$  
$S = (n-1)^{-1} Y' (I - (1/n)J) Y$  
$S_{pl} = \frac{(n_1 - 1 )s_1^2 + (n_2 -  1)s_2^2}{n_1 + n_2 - 2}$ 

In [24]:
Smales <- t(males) %*% (diag(32) - (1/n) * J(32)) %*% males
Sfemal <- t(femal) %*% (diag(32) - (1/n) * J(32)) %*% femal
Spl <- ( (n-1) * Smales + (n-1) * Sfemal ) / ( n+n - 2)

malesbar <- matrix(colSums(males)/n, nrow =4, ncol =1 )
femalbar <- matrix(colSums(femal)/n, nrow = 4, ncol = 1)

a <- solve(Spl) %*% (malesbar - femalbar)
z1 <- t(a) %*% t(males)
z1

[,1]      [,2]      [,3]      [,4]      [,5]      [,6]      [,7]    
[1,] 0.3564512 0.4028941 0.3613887 0.1258607 0.3289938 0.2903616 0.332835
     [,8]      [,9]     [,10]     [,11] [,12]     [,13]     [,14]     [,15]    
[1,] 0.2922329 0.465798 0.3628375 ...   0.2288354 0.4092363 0.3141888 0.4497317
     [,16]     [,17]     [,18]    [,19]     [,20]     [,21]    
[1,] 0.3163051 0.3382836 0.423096 0.4578032 0.2546819 0.3595473

**b)** Calculate t-tests for the individual variables. 

**c)** Compare the results of a) and b) as tothe contribution of the variables to separation of the two groups.

**d)** Find the partial F for each vraiable, as in (8.26), and compare with the standardized coefficients.